onnxruntime/core/graph/slice_elimination.cc

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/graph/rewrite_rule.h"
+#include "core/graph/slice_elimination.h"
+#include "core/graph/graph.h"
+#include "core/graph/op.h"
+#include "core/common/logging/logging.h"
+
+namespace onnxruntime {
+
+Status EliminateSlice::Apply(GraphEditor& graph_editor, Node& node, bool& modified) {
+  if (graph_editor.RemoveSingleInSingleOutNode(node.Index())) {
+    modified = true;
+  }
+
+  return Status::OK();
+}
+
+bool EliminateSlice::SatisfyCondition(const GraphEditor& graph_editor, const Node& node) {
```

Comment:
I'd suggest putting this node elimination rule into one, as all of them are removing one node.